### PR TITLE
Use attr_reader and rename some functions

### PR DESCRIPTION
**Pull request by stackmystack**

### CODE REVIEW

File: lib/oppen/pretty_printer.rb [hunk 1]
```diff
@@ -10,6 +10,44 @@ module Oppen
   # PrettyPrinter class responsible for scanning tokens and passing
   # them to the printer.
   class PrettyPrinter
+    # Ring buffer left indexe.
```
Comment (Amine-Mike) [inline lib/oppen/pretty_printer.rb]: `indexe` is not an english word

File: lib/oppen/pretty_printer.rb [hunk 2]
```diff
@@ -10,6 +10,44 @@ module Oppen
...
+
+    # Total number of spaces needed to print from start of buffer to the left.
+    #
+    # @note Called leftTotal as well in the original paper.
+    attr_reader :left_total
+
+    # @note Called printStack as well in the original paper.
+    attr_reader :print_stack
+
+    # Ring buffer right indexe.
```
Comment (Amine-Mike) [inline lib/oppen/pretty_printer.rb]: Ditto

File: lib/oppen/pretty_printer.rb [hunk 3]
```diff
@@ -93,15 +112,16 @@ def handle_eof
     #
     # @see Token::Begin
     def handle_begin(token)
-      if @scan_stack.empty?
-        @left_total = @right_total = 1
+      if scan_stack.empty?
+        @left_total = 1
+        @right_total = 1
         @left = @right = 0
```
Comment (Amine-Mike) [inline lib/oppen/pretty_printer.rb]: choose between assignation on same line or no (I prefer the first):
```ruby
left_total = right_total = 1
left = right = 0
# Or
left_total = 1
right_total = 1
left = 0
right = 0
```

File: lib/oppen/pretty_printer.rb [hunk 4]
```diff
@@ -126,16 +146,16 @@ def handle_end(token)
     #
     # @see Token::Break
     def handle_break(token)
-      if @scan_stack.empty?
+      if scan_stack.empty?
         @left_total = @right_total = 1
         @left = @right = 0
```
Comment (Amine-Mike) [inline lib/oppen/pretty_printer.rb]: Write the same way as in handle_begin's empty stack condition

**Review by Amine-Mike — CHANGES_REQUESTED** [review]
Watch also you are not passing the CI

File: lib/oppen/pretty_printer.rb (hunk 1 above)
Comment (stackmystack) [reply lib/oppen/pretty_printer.rb]: Hah. I am famous for this. I will install [typos](https://github.com/crate-ci/typos/) in the CI.

File: lib/oppen/pretty_printer.rb (hunk 3 above)
Comment (stackmystack) [reply lib/oppen/pretty_printer.rb]: Good catch. I like them sorted.

**Comment by stackmystack** [comment]
I also suggest we rename `pretty_printer.rb` and its class to `printer.rb` and `Printer`. wdyt?

**Comment by Am1ne77** [comment]
> I also suggest we rename `pretty_printer.rb` and its class to `printer.rb` and `Printer`. wdyt?

**Comment by Amine-Mike** [comment]
> I also suggest we rename `pretty_printer.rb` and its class to `printer.rb` and `Printer`. wdyt?

I agree

File: lib/oppen/pretty_printer.rb (hunk 1 above)
Comment (Amine-Mike) [inline lib/oppen/pretty_printer.rb]: The typo is still present

File: lib/oppen/pretty_printer.rb (hunk 2 above)
Comment (Amine-Mike) [inline lib/oppen/pretty_printer.rb]: Ditto

**Review by stackmystack — COMMENTED** [review]
All reviews addressed.

**Review by Amine-Mike — APPROVED** [review]

**Pull request merged**